trunner/harness/plo.py

```diff
@@ -0,0 +1,288 @@
+from dataclasses import dataclass
+from typing import Callable, Optional, Sequence
+
+import pexpect
+
+from trunner.dut import Dut
+from trunner.text import bold
+from trunner.tools import Phoenixd
+from trunner.types import AppOptions, TestResult
+from .base import HarnessError, IntermediateHarness, Rebooter, TermHarness, VoidHarness
+
+
+class PloError(HarnessError):
+    """Exception thrown by PloInterface class."""
+
+    def __init__(
+        self,
+        msg: Optional[str] = None,
+        output: Optional[str] = None,
+        expected: Optional[str] = None,
+        cmd: Optional[str] = None,
+    ):
+        super().__init__()
+        self.msg = msg
+        self.output = output
+        self.cmd = cmd
+        self.expected = expected
+
+    def __str__(self):
+        err = bold("PLO ERROR: ") + (self.msg if self.msg else "") + "\n"
```

Comment:
I'd use building a list followed by join at the end as in other parts of the code

Reply:
done